src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/CSharpDocumentExcerptService.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Razor;
+using System.Diagnostics;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    internal sealed class CSharpDocumentExcerptService : DocumentExcerptServiceBase
+    {
+        private readonly IRazorSpanMappingService _mappingService;
+
+        private readonly LSPDocumentSnapshot _documentSnapshot;
+
+        public CSharpDocumentExcerptService(
+            IRazorSpanMappingService mappingService,
+            LSPDocumentSnapshot documentSnapshot)
+        {
+            if (mappingService is null)
+            {
+                throw new ArgumentNullException(nameof(mappingService));
+            }
+
+            if (documentSnapshot is null)
+            {
+                throw new ArgumentNullException(nameof(documentSnapshot));
+            }
+
+            _mappingService = mappingService;
+
+            _documentSnapshot = documentSnapshot;
+        }
+
+        internal override async Task<ExcerptResultInternal?> TryGetExcerptInternalAsync(
+            Document document,
+            TextSpan span,
+            ExcerptModeInternal mode,
+            CancellationToken cancellationToken)
+        {
+            var mapped = await _mappingService.MapSpansAsync(document, new[] { span }, cancellationToken).ConfigureAwait(false);
+            if (mapped.Length == 0 || mapped[0].Equals(default(RazorMappedSpanResult)))
```

Comment:
For what cases was `mapped[0].Equals(default(RazorMappedSpanResult))` necessary?

Reply:
`RazorMappedSpanResult` is a struct, so I'm assuming this is just a null check for all intents and purposes. I was following https://github.com/dotnet/aspnetcore-tooling/blame/3acca07e331dabbc83e85e49d675c3aae1e00b75/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorDocumentExcerptService.cs#L56 which had that check. Shouldn't hurt to leave it in IMO.